esmvaltool/diag_scripts/weighting/weighted_temperature_map.py

```diff
@@ -0,0 +1,173 @@
+import xarray as xr
+from cartopy.mpl.ticker import LatitudeFormatter, LongitudeFormatter
+from climwip import (
+    get_diagnostic_filename,
+    get_plot_filename,
+    log_provenance,
+    read_model_data,
+)
+
+from esmvaltool.diag_scripts.shared import run_diagnostic
+from esmvaltool.diag_scripts.weighting.plot_utilities import (
+    calculate_percentiles,
+    read_metadata,
+    read_weights,
+)
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def mapplot(dataarray, cfg, title_pattern, filename_part, ancestors,
+            **colormesh_args):
+    """Visualize weighted temperature."""
+    period = '{start_year}-{end_year}'.format(**read_metadata(cfg)['tas'][0])
+    if 'tas_reference' in read_metadata(cfg).keys():
+        meta = read_metadata(cfg)['tas_reference']
+        period = 'change: {} minus {start_year}-{end_year}'.format(
+            period, **meta[0])
+    metric = cfg['model_aggregation']
+    if isinstance(metric, int):
+        metric = f'{metric}perc'
+    proj = ccrs.PlateCarree(central_longitude=0)
+    figure, axes = plt.subplots(subplot_kw={'projection': proj})
+
+    dataarray.plot.pcolormesh(
+        ax=axes,
+        transform=ccrs.PlateCarree(),
+        levels=9,
+        robust=True,
+        extend='both',
+        **colormesh_args
+        # colorbar size often does not fit nicely
+        # https://stackoverflow.com/questions/18195758/set-matplotlib-colorbar-size-to-match-graph
+        # cbar_kwargs={'fraction': .021}
+    )
+
+    lons = dataarray.lon.values
+    lats = dataarray.lat.values
+    longitude_formatter = LongitudeFormatter()
+    latitude_formatter = LatitudeFormatter()
+    default_xticks = np.arange(np.floor(lons.min()), np.ceil(lons.max()), 10)
+    default_yticks = np.arange(np.floor(lats.min()), np.ceil(lats.max()), 10)
+
+    axes.coastlines()
+    axes.set_xticks(cfg.get('xticks', default_xticks), crs=proj)
+    axes.set_yticks(cfg.get('yticks', default_yticks), crs=proj)
+    axes.xaxis.set_ticks_position('both')
+    axes.yaxis.set_ticks_position('both')
+    axes.xaxis.set_major_formatter(longitude_formatter)
+    axes.yaxis.set_major_formatter(latitude_formatter)
+    axes.set_xlabel('')
+    axes.set_ylabel('')
+
+    title = title_pattern.format(metric=metric, period=period)
+    axes.set_title(title)
+
+    filename_plot = get_plot_filename(filename_part, cfg)
+    figure.savefig(filename_plot, dpi=300, bbox_inches='tight')
+    plt.close(figure)
+
+    filename_data = get_diagnostic_filename(filename_part, cfg, extension='nc')
+    dataarray.to_netcdf(filename_data)
+
+    log_provenance(title, filename_plot, cfg, ancestors)
+    log_provenance(title, filename_data, cfg, ancestors)
+
+
+def visualize_and_save_temperature(temperature: 'xr.DataArray', cfg: dict,
+                                   ancestors: list):
+    """Wrap mapplot: absolute temperature."""
+    title_pattern = '\n'.join(
+        ['Weighted {metric} temperature', r'{period} ($\degree$C)'])
```

Comment:
Perhaps something like this could also be an alternative:

```suggestion
    title_pattern = (
        'Weighted {metric} temperature'
        r'{period} ($\degree$C)'
    )
```

Reply:
Ah sorry I did not see this before! Changed this now independent of your suggestion.